code/lib/csf-tools/src/CsfFile.ts

```diff
@@ -234,7 +234,8 @@ export class CsfFile {
     return node;
   }
 
-  parse() {
+  // TODO: is there a better way to type this correctly?
+  parse(): CsfFile & { meta: StaticMeta; stories: IndexedStory[] } {
```

Comment:
Would like some input here @kasperpeulen @tmeasday

There's no new implementation, [I just added types to our existing indexers](https://github.com/storybookjs/storybook/pull/23582/files#diff-e3d50a086df937d7c8f41d5819f0e6d8368961c112866cc291cfe057a4fafe23R132-R133) which surfaced this type error.
The problem is, that:

1. Most indexers return the result `CsfFile.parse()` as the result of the indexer.
2. An indexer needs to return an `IndexedCsfFile`
3. `CsfFile.parse()` returns `CsfFile`, not `IndexedCsfFile`
4. The difference between the two is that:
	1. `meta` can be `undefined` in `CsfFile`, it can't in `IndexedCsfFile`
	2. `stories.*.name` can be `undefined` in `CsfFile` (`StaticStory`), it can't in `IndexedCsfFile` (`IndexedStory`)
5. The implementation of `CsfFile.parse()` ensures that both of these constraints are met, but TypeScript of course doesn't know that because 3.

My solution here makes TypeScript happy, but it doesn't feel like a type-safe way to solve this?

Reply:
Is there a reason you didn't just change it to return an `IndexedCSFFile` (I assume that's what you meant?)

Reply:
`IndexedCSFFile` has less properties than `CsfFile`, so other callsites that uses `parse()` won't accept that. But I can change it to `CsfFile & IndexedCSFFile` which should be okay I guess.

Reply:
I don't really know the difference between the two types but is another option to just add the fields to `CsfFile` if that's what's returned by the `CsfFile` library?

Reply:
The issue is that `CsfFile` already have `meta` and `stories`, but they can be `undefined` because they are not set during construction, only after `parse()` is called.